package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -78,11 +86,14 @@ def layers_mapping(self):  # noqa: PLR0912
 
         self._layers_mapping = {}
 
-        # Temporary try/except block so the Kedro develop branch can work with Viz.
-        try:
-            datasets = self._catalog._data_sets
-        except Exception:  # noqa: BLE001 # pragma: no cover
-            datasets = self._catalog._datasets
+        if IS_KEDRODATACATALOG and isinstance(self._catalog, KedroDataCatalog):
+            datasets = self._catalog.list()
+        else:
+            # Temporary try/except block so the Kedro develop branch can work with Viz.
+            try:
+                datasets = self._catalog._data_sets
+            except Exception:  # noqa: BLE001 # pragma: no cover
+                datasets = self._catalog._datasets
```

Comment:
Is this still needed? It mentions the kedro develop branch, but I'm guessing that was related to pre-0.19?

Reply:
Good spot, this branch was released a while ago I will remove this block.

Reply:
Actually we still need that try/except because in older Kedro versions (like 0.18.3), as CI started failing for python >3.10.